tests/apidoc_request_encoding_test.go

```diff
@@ -0,0 +1,28 @@
+package tests
+
+import (
+	"testing"
+
+	"encoding/json"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestRequestEncodingFails(t *testing.T) {
+	deployment := Deploy(t, "request_encoding", b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	testString := `{ "test": "a' . ` + "\x81" + ` . '" }`
+	// sytest: POST rejects invalid utf-8 in JSON
+	t.Run("POST rejects invalid utf-8 in JSON", func(t *testing.T) {
+		res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "register"}, json.RawMessage(testString))
```

Comment:
`MustDo` will:

> // MustDo is the same as Do but fails the test if the response is not 2xx

We _want_ the test to return 400, so you cannot use `MustDo` here. Use `Do` instead.

Reply:
Oops, Thanks for noticing, will fix this asap.